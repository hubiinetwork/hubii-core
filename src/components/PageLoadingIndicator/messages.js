/*
 * PageLoadingIndicator Messages
 *
 * This contains all the text for the PageLoadingIndicator component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'app.components.PageLoadingIndicator.header',
    defaultMessage: 'Fetching {pageType} {id}',
  },
});
