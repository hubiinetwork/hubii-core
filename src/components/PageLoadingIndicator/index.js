/**
*
* PageLoadingIndicator
*
*/

import React from 'react';
import PropTypes from 'prop-types';

import { Loader } from 'semantic-ui-react';

import { FormattedMessage } from 'react-intl';
import messages from './messages';

function PageLoadingIndicator(props) {
  return (
    <div>
      <Loader active={props.loading} size="huge">
        <FormattedMessage
          {...messages.header}
          values={{ pageType: props.pageType, id: props.id }}
        />
      </Loader>
    </div>
  );
}

PageLoadingIndicator.propTypes = {
  id: PropTypes.string.isRequired,
  pageType: PropTypes.string.isRequired,
  loading: PropTypes.bool.isRequired,
};

export default PageLoadingIndicator;
