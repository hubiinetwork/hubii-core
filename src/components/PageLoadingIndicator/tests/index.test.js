import React from 'react';
import { shallow } from 'enzyme';

import PageLoadingIndicator from '../index';

describe('<PageLoadingIndicator />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<PageLoadingIndicator id={'12'} pageType={'FooPage'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
