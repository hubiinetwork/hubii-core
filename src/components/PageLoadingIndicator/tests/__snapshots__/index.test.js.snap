// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLoadingIndicator /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageLoadingIndicator
    id="12"
    pageType="FooPage"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Loader
        active={true}
        size="huge"
>
        <FormattedMessage
                defaultMessage="Fetching {pageType} {id}"
                id="app.components.PageLoadingIndicator.header"
                values={
                        Object {
                                "id": "12",
                                "pageType": "FooPage",
                              }
                }
        />
</Loader>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": true,
        "children": <FormattedMessage
          defaultMessage="Fetching {pageType} {id}"
          id="app.components.PageLoadingIndicator.header"
          values={
                    Object {
                              "id": "12",
                              "pageType": "FooPage",
                            }
          }
/>,
        "size": "huge",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultMessage": "Fetching {pageType} {id}",
          "id": "app.components.PageLoadingIndicator.header",
          "values": Object {
            "id": "12",
            "pageType": "FooPage",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Loader
          active={true}
          size="huge"
>
          <FormattedMessage
                    defaultMessage="Fetching {pageType} {id}"
                    id="app.components.PageLoadingIndicator.header"
                    values={
                              Object {
                                        "id": "12",
                                        "pageType": "FooPage",
                                      }
                    }
          />
</Loader>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": true,
          "children": <FormattedMessage
            defaultMessage="Fetching {pageType} {id}"
            id="app.components.PageLoadingIndicator.header"
            values={
                        Object {
                                    "id": "12",
                                    "pageType": "FooPage",
                                  }
            }
/>,
          "size": "huge",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultMessage": "Fetching {pageType} {id}",
            "id": "app.components.PageLoadingIndicator.header",
            "values": Object {
              "id": "12",
              "pageType": "FooPage",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
